AWSAPIGatewayTests/AWSLambdaMicroserviceClient.m

```diff
@@ -72,16 +86,51 @@ + (instancetype)defaultClient {
     return _defaultClient;
 }
 
++ (void)registerClientWithConfiguration:(AWSServiceConfiguration *)configuration forKey:(NSString *)key {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        _serviceClients = [AWSSynchronizedMutableDictionary new];
+    });
+    [_serviceClients setObject:[[AWSLambdaMicroserviceClient alloc] initWithConfiguration:configuration]
+                        forKey:key];
+}
+
++ (instancetype)clientForKey:(NSString *)key {
+    @synchronized(self) {
+        AWSLambdaMicroserviceClient *serviceClient = [_serviceClients objectForKey:key];
+        if (serviceClient) {
+            return serviceClient;
+        }
+
+        AWSServiceInfo *serviceInfo = [[AWSInfo defaultAWSInfo] serviceInfo:AWSInfoClientKey
+                                                                     forKey:key];
+        if (serviceInfo) {
+            AWSServiceConfiguration *serviceConfiguration = [[AWSServiceConfiguration alloc] initWithRegion:serviceInfo.region
+                                                                                        credentialsProvider:serviceInfo.cognitoCredentialsProvider];
+            [AWSLambdaMicroserviceClient registerClientWithConfiguration:serviceConfiguration
+                                                    forKey:key];
+        }
+
+        return [_serviceClients objectForKey:key];
+    }
+}
+
++ (void)removeClientForKey:(NSString *)key {
+    [_serviceClients removeObjectForKey:key];
+}
+
+- (instancetype)init {
+    @throw [NSException exceptionWithName:NSInternalInconsistencyException
+                                   reason:@"`- init` is not a valid initializer. Use `+ defaultClient` or `+ clientForKey:` instead."
+                                 userInfo:nil];
+    return nil;
+}
+
 - (instancetype)initWithConfiguration:(AWSServiceConfiguration *)configuration {
     if (self = [super init]) {
         _configuration = [configuration copy];
-        NSString *filePath = [[NSBundle bundleForClass:[self class]] pathForResource:@"credentials"
-                                                                              ofType:@"json"];
-        NSDictionary *credentialsJson = [NSJSONSerialization JSONObjectWithData:[NSData dataWithContentsOfFile:filePath]
-                                                                        options:NSJSONReadingMutableContainers
-                                                                          error:nil];
 
-        NSString *URLString = credentialsJson[@"apiEndpoint"];
```

Comment:
Should this be part of the credentials.json file?

Reply:
This was downloaded verbatim from the API Gateway site. I'd prefer to leave it as-is so we get a better test of what customers are actually dealing with. It's not sensitive information, inasmuch as it simply exposes an echo API that we can protect with appropriate auth methods as needed.